modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -273,9 +275,25 @@ package object widgets {
     encoded
   } else Nil
 
+  def fromDFToPoint(df:DataFrame, maxPoints:Int):Seq[MagicRenderPoint] = {
+    val rows = df.take(maxPoints)
+    if (rows.nonEmpty) {
+      val points = df.schema.toList.head match {
```

Comment:
I know this patch is a first-cut but magicrenderpoint is just bad news here; only the schema should be used (not reflection).  Df.toJSON is your friend.